apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/TimedBufferingConsumer.cs

```diff
@@ -14,28 +14,31 @@
 
 using Google.Api.Gax;
 using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Google.Cloud.Diagnostics.Common
 {
     /// <summary>
     /// A <see cref="IFlushableConsumer{T}"/> that will automatically flush the buffer after a 
     /// given amount of time.
     /// </summary>
-    internal class TimedBufferingConsumer<T> : FlushableConsumerBase<T>
+    internal class TimedBufferingConsumer<T> : IFlushableConsumer<T>
     {
-        /// <summary>
-        /// The buffered items. This is not readonly as it is replaced when the buffer is flushed.
-        /// </summary>
-        private List<T> _items = new List<T>();
+        /// <summary>The consumer to flush to.</summary>
+        private readonly IConsumer<T> _consumer;
+
+        private BlockingCollection<T> _items = new BlockingCollection<T>();
```

Comment:
Can we call this something else? I've been really confused when reviewing this because we're getting an enumerator from a parameter called `items` and adding values to `_items`.

Maybe `_bufferedItems` or `_receivedItems`?